### ISSUE TITLE
SFTP fails on Easylink server

### ISSUE THREAD

**Issue by stetskevych** [lavv17/lftp#358]
`$ lftp -p22 sftp://vanconnect.easylink.com
lftp vanconnect.easylink.com:~> user xxxxxx yyyyyy
lftp xxxxxx@vanconnect.easylink.com:~> debug
lftp xxxxxx@vanconnect.easylink.com:~> ls
---- Running connect program (ssh -a -x -s -l xxxxxx -p 22 vanconnect.easylink.com sftp)
---> sending a packet, length=5, type=1(INIT), id=0
<--- Password authentication
`ls' at 0 [Connecting...]
`

When I run the connect program directly, it prompts for the password correctly and launches the sftp subsystem (when run in verbose mode):
`$ ssh -a -x -s -l xxxxxx -p 22 vanconnect.easylink.com sftp
Password authentication
xxxxxx's password`

Any hints to fix lftp, please?

**Comment by lavv17** [comment]
For a reason your ssh command uses an unusual password prompt which is not recognized by lftp.

**Comment by stetskevych** [comment]
I agree, but it's not my server, it's one large US service provider. And I
cannot tell them to change the prompt, so I thought there could be an
option on lftp side to adapt.

By the way, the `sftp` command can connect normally to that server.

**Comment by lavv17** [comment]
I think the "sftp" command does not send the password to ssh, but allows it to read it by itself.

**Comment by lavv17** [comment]
Please tell me where the cursor blinks when ssh asks for the password.

**Comment by stetskevych** [comment]
`$ ssh -a -x -s -l xxxxxx -p 22 vanconnect.easylink.com sftp
Password authentication
xxxxxx's password[]
`
The cursor is [].

**Comment by stetskevych** [comment]
You can actually try the ssh command for yourself. It works from any IP.

**Comment by oliparcol** [comment]
Hello,

I have the exact same issue with a different site probably using the same sftp server. Is there any workaround ? 

Thanks for your help,

**Comment by lavv17** [comment]
There is no workaround, just use the ftps scheme.

чт, 14 дек. 2017, 21:30 Olivier Parent-Colombel <notifications@github.com>:

**Comment by CarstenGrohmann** [comment]
We are also affected. The password prompt just differs by a missing colon like the initial report shows.

Please adapt the detection of the SSH password prompt.

**Remote application Cleo VLtrader v5.4:**
```
# ssh -a -x -v MYUSER@vltrader.domain.com
OpenSSH_7.4p1, OpenSSL 1.0.2k-fips  26 Jan 2017
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 58: Applying options for *
debug1: Connecting to vltrader.domain.com [10.10.10.10] port 22.
debug1: Connection established.
[...]
debug1: Next authentication method: keyboard-interactive
Password authentication
MYUSER's password
```

**Remote application OpenSSH_7.4p1:**
```
# ssh -a -x -v MYUSER@openssh.domain.com
OpenSSH_7.4p1, OpenSSL 1.0.2k-fips  26 Jan 2017
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 58: Applying options for *
debug1: Connecting to openssh.domain.com [10.10.10.11] port 22.
debug1: Connection established.
[...]
debug1: Next authentication method: password
MYUSER@openssh.domain.com's password:
```

Thank you,
Carsten
